testing/e2e/skr-tester/pkg/kcp/client.go

```diff
@@ -0,0 +1,101 @@
+package kcp
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+type KCPConfig struct {
+	AuthType          string
+	Host              string
+	IssuerURL         string
+	GardenerNamespace string
+	Username          string
+	Password          string
+	ClientID          string
+	ClientSecret      string
+	KubeConfigApiUrl  string
+}
+
+type KCPClient struct {
+	Config *KCPConfig
+}
+
+func NewKCPConfig() *KCPConfig {
+	return &KCPConfig{
+		AuthType:          getEnvOrThrow("KCP_AUTH_TYPE"),
+		Host:              getEnvOrThrow("KCP_KEB_API_URL"),
+		IssuerURL:         getEnvOrThrow("KCP_OIDC_ISSUER_URL"),
+		ClientID:          getEnvOrThrow("KCP_OIDC_CLIENT_ID"),
+		ClientSecret:      getEnvOrThrow("KCP_OIDC_CLIENT_SECRET"),
+		GardenerNamespace: getEnvOrThrow("KCP_GARDENER_NAMESPACE"),
+		Username:          getEnvOrThrow("KCP_TECH_USER_LOGIN"),
+		Password:          getEnvOrThrow("KCP_TECH_USER_PASSWORD"),
+		KubeConfigApiUrl:  getEnvOrThrow("KCP_KUBECONFIG_API_URL"),
+	}
+}
+
+func NewKCPClient() *KCPClient {
+	client := &KCPClient{}
+	if clientSecret := os.Getenv("KCP_OIDC_CLIENT_SECRET"); clientSecret != "" {
+		client.Config = NewKCPConfig()
+		client.WriteConfigToFile()
+	}
+	return client
```

Comment:
The method returns uninitialized client without `KCP_OIDC_CLIENT_SECRET` value

Reply:
That's the point. When we run kcp cli on our machines, we don't need this configuration here. It's only initialised for the technical user in the pipeline.